pymc3/tests/test_distributions_random.py

```diff
@@ -498,43 +425,177 @@ class TestMoyal(BaseTestCases.BaseTestCase):
+        )
 
-    def test_probability_vector_shape(self):
-        """Check that if a 2d array of probabilities are passed to categorical correct shape is returned"""
-        p = np.ones((10, 5))
-        assert pm.Categorical.dist(p=p).random().shape == (10,)
-        assert pm.Categorical.dist(p=p).random(size=4).shape == (4, 10)
-        p = np.ones((3, 7, 5))
-        assert pm.Categorical.dist(p=p).random().shape == (3, 7)
-        assert pm.Categorical.dist(p=p).random(size=4).shape == (4, 3, 7)
+    def _pymc_params_match_aesara_rv_ones(
+        self, pymc_dist_output, expected_aesara_params, decimal=6
+    ):
+        aesera_dist_inputs = self.get_inputs_from_apply_node_outputs(pymc_dist_output)[3:]
+        assert len(expected_aesara_params) == len(aesera_dist_inputs)
+        for (expected_name, expected_value), actual_variable in zip(
+            expected_aesara_params, aesera_dist_inputs
+        ):
+            assert_almost_equal(expected_value, actual_variable.eval(), decimal=decimal)
+
+    def _pymc_sample_matches_aeasara_rv_one(
+        self, pymc_dist_output, pymc_dist, expected_aesara_params, size, decimal
+    ):
+        sample = pymc_dist.rv_op(
+            *[p[1] for p in expected_aesara_params],
+            size=size,
+            rng=aesara.shared(self.get_random_state()),
+        )
+        assert_array_almost_equal(pymc_dist_output.eval(), sample.eval(), decimal=decimal)
 
+    def test_normal(self):
+        params = [("mu", 5.0), ("sigma", 10.0)]
+        self._compare_pymc_sampling_with_aesara_one(params, list(params), pm.Normal)
 
-@pytest.mark.skip(reason="This test is covered by Aesara")
-class TestDirichlet(SeededTest):
-    @pytest.mark.parametrize(
-        "shape, size",
-        [
-            ((2), (1)),
-            ((2), (2)),
-            ((2, 2), (2, 100)),
-            ((3, 4), (3, 4)),
-            ((3, 4), (3, 4, 100)),
-            ((3, 4), (100)),
-            ((3, 4), (1)),
-        ],
-    )
-    def test_dirichlet_random_shape(self, shape, size):
-        out_shape = to_tuple(size) + to_tuple(shape)
-        assert pm.Dirichlet.dist(a=np.ones(shape)).random(size=size).shape == out_shape
+    def test_uniform(self):
+        params = [("lower", 0.5), ("upper", 1.5)]
+        self._compare_pymc_sampling_with_aesara_one(params, list(params), pm.Uniform)
+
+    def test_half_normal(self):
+        params, expected_aesara_params = [("sigma", 10.0)], [("mean", 0), ("sigma", 10.0)]
+        self._compare_pymc_sampling_with_aesara_one(params, expected_aesara_params, pm.HalfNormal)
+
+    def test_beta_alpha_beta(self):
+        params = [("alpha", 2.0), ("beta", 5.0)]
+        self._compare_pymc_sampling_with_aesara_one(params, list(params), pm.Beta)
+
+    def test_beta_mu_sigma(self):
+        params = [("mu", 0.5), ("sigma", 0.25)]
+        expected_alpha, expected_beta = pm.Beta.get_alpha_beta(mu=params[0][1], sigma=params[1][1])
```

Comment:
Small nitpick, would it be a lot of trouble to write the tests expecting a dictionary of `{"parameter_name": parameter_value}` instead of the list of tuples? I think it makes it easier to write / read, and it seems that in the test function you are converting it to a dict anyway. Dictionaries are ordered from 3.6, which is what we support, so order wouldn't be an issue.

Reply:
Also maybe with this refactoring we can avoid the `params, list(params)` in the function call.

Reply:
Good idea, about using a dict, given its current ordered nature! I live in the past :-). yeah `list(params)` is very ugly, sorry for that, worse case I will replace it with `copy.copy`